Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelper.java

```diff
@@ -0,0 +1,202 @@
+
+    }
+    
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_Outbound() {
+
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+        eventA.setDirection("outbound");
+        AdhocQueryRequestMessageType msg = new AdhocQueryRequestMessageType();
+        AdhocQueryRequest adhocQueryRequest = new AdhocQueryRequest();
+        SlotType1 slotType1 = new SlotType1();
+        slotType1.setName("$XDSDocumentEntryPatientId");
+
+        ValueListType valueListType = new ValueListType();
+        valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+        slotType1.setValueList(valueListType);
+        AdhocQueryType adhocQueryType = new AdhocQueryType();
+        adhocQueryType.getSlot().add(slotType1);
+
+        adhocQueryRequest.setAdhocQuery(adhocQueryType);
+        msg.setAdhocQueryRequest(adhocQueryRequest);
+        AssertionType assertionTypeTest = new AssertionType();
+        assertionTypeTest.setMessageId("12345566");
+        HomeCommunityType homeCom = new HomeCommunityType();
+        homeCom.setHomeCommunityId("urn:oid:1.1.1.1.1.1.1");
+        assertionTypeTest.setHomeCommunity(homeCom);
+        msg.setAssertion(assertionTypeTest);
+        eventA.setMessage(msg);
+
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        verifycheckOutboundPolicy(result);
+
+    }
+    
+    private void verifycheckOutboundPolicy(CheckPolicyRequestType actualCheckPolicy) {
+        // Verify result
+        assertNotNull(actualCheckPolicy);
+        // AssertAssertion to make sure nothing change
+        AssertionType assertTypeResult = actualCheckPolicy.getAssertion();
+
+        assertEquals("messageId should not change", "12345566", assertTypeResult.getMessageId());
+        // Assert Resource Session
+        assertEquals("urn:oasis:names:tc:xacml:1.0:resource:resource-id", actualCheckPolicy.getRequest().getResource()
+            .get(0).getAttribute().get(1).getAttributeId());
+        String expectResourceId = "amp;26.489.22";
+        assertEquals(expectResourceId, actualCheckPolicy.getRequest().getResource().get(0).getAttribute().get(0)
+            .getAttributeValue().get(0).getContent().get(0));
+
+        assertEquals("DocumentQueryOut", actualCheckPolicy.getRequest().getAction().getAttribute().get(0)
+            .getAttributeValue().get(0).getContent().get(0));
+        SubjectType subjectType = actualCheckPolicy.getRequest().getSubject().get(0);
+        AttributeType attributeType = subjectType.getAttribute().get(1);
+        assertEquals("http://www.hhs.gov/healthit/nhin#HomeCommunityId", attributeType.getAttributeId());
+        assertEquals("http://www.w3.org/2001/XMLSchema#anyURI", attributeType.getDataType());
+        assertEquals("urn:oid:1.1.1.1.1.1.1", attributeType.getAttributeValue().get(0).getContent().get(0));
+    }
+  
+
+    private void verifycheckInboundPolicy(CheckPolicyRequestType actualCheckPolicy) {
```

Comment:
verifycheckInboundPolicy and verifycheckOutboundPolicy are almost the same except DocumentQueryOut vs DocumentQueryIn.  Try to refactor so we don't duplicate method in here